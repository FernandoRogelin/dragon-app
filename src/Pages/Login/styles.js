import styled from "styled-components";

export const Wrapper = styled.section`
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Content = styled.div`
  width: 360px;
  padding: 20px;
  border-radius: 10px;
  box-sizing: border-box;
  box-shadow: 0 5px 30px rgba(0, 0, 0, 0.2);
`;

export const Title = styled.h1`
  margin-bottom: 20px;
  color: ${({ theme }) => theme.pigeonPost};
`;

export const Body = styled.div`
  position: relative;
  margin-bottom: 20px;
`;

export const Error = styled.span`
  left: 0;
  bottom: -15px;
  font-size: 12px;
  position: absolute;
  color: ${({ theme }) => theme.amaranth};
`;
