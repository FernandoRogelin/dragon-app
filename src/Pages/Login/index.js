import React from "react";

import Routes from "../../Routes";
import Conta from "../../Utils/conta";
import { useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";
import { InputForm, Button } from "../../Components";

import { Wrapper, Content, Title, Body, Error } from "./styles";

export default function Login() {
  let history = useHistory();

  const { handleSubmit, register, errors, formState } = useForm({
    mode: "onChange",
  });

  async function onSubmit(values) {
    if (values.email === Conta.email && values.password === Conta.password) {
      localStorage.setItem("codUser", "4554");
      history.push(Routes.dragons);
    }
  }

  return (
    <Wrapper>
      <Content>
        <Title>Acesse a sua conta</Title>
        <form onSubmit={handleSubmit(onSubmit)}>
          <Body>
            <InputForm
              type="email"
              name="email"
              testId="input-email"
              error={errors.email}
              placeholder="E-mail"
              ariaLabel="formulario email"
              refInput={register({
                required: "Campo obrigatório.",
                pattern: {
                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                  message: "Formato do e-mail errado.",
                },
              })}
            />
            {errors.email && formState.touched.email && (
              <Error>{errors.email.message}</Error>
            )}
          </Body>
          <Body>
            <InputForm
              type="password"
              name="password"
              placeholder="Senha"
              testId="input-password"
              ariaLabel="formulario senha"
              error={errors.password}
              refInput={register({
                required: "Campo obrigatório.",
                validate: (value) => value.length > 5,
              })}
            />
            {errors.password && formState.touched.password && (
              <Error>{errors.password.message}</Error>
            )}
          </Body>
          <Button
            type="submit"
            aria-label="logar"
            data-testid="button-login"
            disabled={!formState.isValid || formState.isSubmitting}
          >
            Entrar
          </Button>
        </form>
      </Content>
    </Wrapper>
  );
}
