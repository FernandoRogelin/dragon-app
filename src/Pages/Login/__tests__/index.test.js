import React from "react";
import { render, fireEvent, act, wait } from "@testing-library/react";

import Login from "..";
const promise = Promise.resolve();

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe("Login page", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Snapshot test", async () => {
    const { container } = render(<Login />);

    expect(container.firstChild).toMatchSnapshot();

    await act(() => promise);
  });

  test("Login account", async () => {
    const { getByTestId } = render(<Login />);

    const inputEmail = getByTestId("input-email");
    const inputPassword = getByTestId("input-password");

    await wait(() =>
      fireEvent.change(inputEmail, {
        target: { value: "fernando.rogelin@hotmail.com" },
      })
    );

    await wait(() =>
      fireEvent.change(inputPassword, { target: { value: "123456" } })
    );

    await fireEvent.click(getByTestId("button-login"));

    await act(() => promise);
  });
});
