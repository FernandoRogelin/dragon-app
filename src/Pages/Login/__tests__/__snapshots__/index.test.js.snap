// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Snapshot test 1`] = `
<section
  class="sc-bdnylx hIQIzg"
>
  <div
    class="sc-gtssRu eKPwjv"
  >
    <h1
      class="sc-dlnjPT jxsYXH"
    >
      Acesse a sua conta
    </h1>
    <form
      id="formulario"
    >
      <div
        class="sc-hKFyIo eLPcgC"
      >
        <input
          aria-labelledby="formulario email"
          class="styles__Input-sc-1tlf70t-0 iWnGNM"
          data-testid="input-email"
          name="email"
          placeholder="E-mail"
          type="email"
          value=""
        />
      </div>
      <div
        class="sc-hKFyIo eLPcgC"
      >
        <input
          aria-labelledby="formulario senha"
          class="styles__Input-sc-1tlf70t-0 iWnGNM"
          data-testid="input-password"
          name="password"
          placeholder="Senha"
          type="password"
          value=""
        />
      </div>
      <button
        aria-label="logar"
        class="Button__button-a83c5q-0 kzEeEq"
        data-testid="button-login"
        type="submit"
      >
        Entrar
      </button>
    </form>
  </div>
</section>
`;
